src/pages/storage/CustomVolumeSelectModal.tsx

```diff
@@ -51,20 +62,46 @@ const CustomVolumeSelectModal: FC<Props> = ({
   const headers = [
     { content: "Name" },
     { content: "Pool" },
+    ...(isClustered ? [{ content: "Location" }] : []),
     { content: "Content type" },
     { content: "Used by" },
     { "aria-label": "Actions", className: "actions" },
   ];
 
+  const isInstance = formik.values.entityType === "instance";
+  const instanceLocation =
+    (formik as FormikProps<EditInstanceFormValues>).values.location ??
+    (formik as FormikProps<CreateInstanceFormValues>).values.target ??
+    "any";
+
   const rows = isFetching
     ? []
     : volumes
         .sort((a, b) => (a.created_at > b.created_at ? -1 : 1))
         .map((volume) => {
-          const selectVolume = () => handleSelect(volume);
+          const isLocalVolume = !!volume.location; // if location is set, it's a local volume, otherwise a remote volume
+          const isDisabled =
+            isClustered &&
+            isInstance &&
+            isLocalVolume &&
+            instanceLocation !== volume.location;
+
+          const disableReason = isDisabled
+            ? `Mismatch of instance location (${instanceLocation}) and local volume location (${volume.location}). `
```

Comment:
maybe a bit easier to understand?
```suggestion
            ? `Instance location (${instanceLocation}) does not match local volume location (${volume.location}). `
```

Reply:
good idea.